src/main/java/ru/practicum/shareit/item/repository/ItemRepository.java

```diff
@@ -0,0 +1,44 @@
+package ru.practicum.shareit.item.repository;
+
+import org.springframework.stereotype.Repository;
+import ru.practicum.shareit.item.model.Item;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+@Repository
+public class ItemRepository {
+    private final Map<Long, Item> items = new HashMap<>();
+    private final AtomicLong idGenerator = new AtomicLong();
+
+    public Item save(Item item) {
+        if (item.getId() == null) {
+            item.setId(idGenerator.incrementAndGet());
+        }
+        items.put(item.getId(), item);
+        return item;
+    }
+
+    public Optional<Item> findById(Long id) {
+        return Optional.ofNullable(items.get(id));
+    }
+
+
+    public List<Item> findByOwner(Long ownerId) {
+        return items.values().stream().filter(item -> item.getOwnerId().equals(ownerId)).collect(Collectors.toList());
```

Comment:
Чтобы сделать стримы более читаемыми их принято разбивать по строкам:
```suggestion
        return items.values().stream()
            .filter(item -> item.getOwnerId().equals(ownerId))
            .collect(Collectors.toList());
```